src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,21 +171,21 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager();
+                string aadIssuer = effectiveConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
+                if (IsValidIssuer(aadIssuer, tenantId, issuer))
                 {
-                    if (AadIssuerV2 == null)
-                        AadIssuerV2 = CreateConfigManager(AadAuthorityV2).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
-                        return issuer;
+                    // todo: set LKG only token is valid
+                    effectiveConfigurationManager.LastKnownGoodConfiguration = new OpenIdConnectConfiguration() { Issuer = aadIssuer };
+                    return issuer;
                 }
-                else
+                else if (validationParameters.ValidateWithLKG && effectiveConfigurationManager.LastKnownGoodConfiguration != null)
```

Comment:
Why not check for LKG validity?

Reply:
AadIssuerValidator's configurationManager(v1 or v2) is synced with tvp.ConfigurationManager by validateWithLKG, they share same LKG lifetime. We already checked the LKG validity in handler.

Reply:
Since they are separate instances, I don't understand how they are synced.

Reply:
Do we want to support LKG only when coming through JsonWebHandler or when AadIssuerValidator is used separately as well?